Medbay Oxigen Assistant

SagaStars opened the pull request:
- Añade el M.O.A (Medbay Oxigen Assistant) curara un 16 de daño de sofocación tambien estabilizando el ritmo cardiaco y dependera de una powercell. Usando un quimico especial que no puede ser obtenible de ninguna manera ( Tarda como 5 minutos en descargarse con una bateria inicial ).
- El M.O.A ya aparecera por defecto en medbay, con una bateria claro.

## Why It's Good For The Game
   La principal razon del objeto es que servira como una alternativa a dejar a un tripulante sin pulmon 15 minutos en cryo mientras que operar a un mono humanizado para sacar un pulmon y transplantarselo, dejando al jugador poder hablar por la radio, responder pdas, ect. 

   En resumen un estabilizador que gaste energia en vez de quimicos de manera efectiva. Que hara el trabajo de medbay un poco mas comodo a la hora de dejar en espera a pacientes o tratar a distintos a la vez.

![](https://cdn.discordapp.com/attachments/643316781980712960/643316803438903306/unknown.png)

![](https://cdn.discordapp.com/attachments/643316781980712960/643316972725075978/unknown.png)


## Changelog
🆑 
add: nuevo equipamiento para medbay (moa)
add: el quimico que usa el moa
🆑

SagaStars committed:
El papa de los commits.

SagaStars committed:
bruh

- Efectua el mapmerge

SagaStars commented:
No tengo ni la menor idea de por que el 2º commit no esta haciendo nada.

SagaStars commented:
> No tengo ni la menor idea de por que el 2º commit no esta haciendo nada.

Bruh, sobrescribio los cambios.

Evankhell561 commented:
¿Tiene algo negativo el ir por ahi con uno de esos?
¿Si no es así, esto me permitiria ir por ahi corriendo sin pulmones?

Evankhell561 commented on code/HISPANIA/modules/reagents/reagent_containers/moa.dm:
```diff
@@ -0,0 +1,175 @@
+
+/obj/item/reagent_containers/moa/proc/deductcharge(var/chrgdeductamt)
+	if(bcell)
+		if(bcell.charge < (hitcost+chrgdeductamt)) // If after the deduction the baton doesn't have enough charge for a stun hit it turns off.
```
> If after the deduction the baton doesn't have enough charge for a stun hit it turns off.

¿El Baton, cual baton?

SagaStars replied:
cHHUcHAAAAAAAAAAAAAA

Evankhell561 commented on code/HISPANIA/modules/reagents/reagent_containers/moa.dm:
```diff
@@ -0,0 +1,175 @@
+			update_icon()
+
+/obj/item/reagent_containers/moa/attack_self(mob/user)
+	..()
```
si lo unico que harás deficinedo el attack_self de éste objeto es:
`..()`
entonces no necesitas esa ninguna de estas lineas de código.

Evankhell561 replied:
Tengo entendido que  `..()` es para llamar a los comportamientos viejos, osea que estás haciendo una deficinion nueva solo para que tu objeto se comporte como se iba a comportar si no hacias una definición nueva de attack_self().

SagaStars replied:
H U H

SagaStars committed:
yikes

Evankhell561 commented:
te recomiendo actualizar tu master, copiar el mapa de tu master, actualizar ésta branch con tu master
y luego pegar manualmente el mapa de tu master en ésta branch. Luego, puedes modificar el mapa de nuevo.

Commit:
Revert "bruh"

This reverts commit 0065bd33dbbdae74de7e7b6b06e8ab9c57b5877a.

Commit:
Revert "yikes"

This reverts commit 231b65c52098e3352974cf51c31ab91fdb08c43c.

Commit:
Intento de arreglo

SagaStars commented:
> ¿Tiene algo negativo el ir por ahi con uno de esos?
> ¿Si no es así, esto me permitiria ir por ahi corriendo sin pulmones?

Pues... Es demasiado grande como para guardarlo en una duffelbag, eso cuenta?

Commit:
Arreglado todito y evan.

- mergeado el dm, tuve problemas instalando python debido a la compatibilidad de visual studio, todo listo.
- se cumplieron las sugerencias de evan

Evankhell561 commented on code/HISPANIA/modules/reagents/reagent_containers/moa.dm:
```diff
@@ -0,0 +1,172 @@
+	amount_per_transfer_from_this = 1
+	var/status = 3
+	var/hitcost = 64
+	var/obj/item/stock_parts/cell/bcell = null
```
¿No queria molestarte porque tenias varios conflictos pero puedes reemplazar todas las "bcell" por "cell" en todo tu codigo? En paradise ya no queda ninguna referencia a los "bcell". Esto no tendrá ningun efecto real pero ayuda a tener un codigo más entendible y más fácil de manejar. En esta linea por ejemplo pasas de  var/obj/item/stock_parts/cell/bcell = null a 
var/obj/item/stock_parts/cell/cell = null

SagaStars replied:
Tambien con los bcell = null?

Evankhell561 replied:
>Tambien con los bcell = null?

Cada vez que veas que tu codigo dice "bcell", lo borras y pones "cell" en su lugar.  Así que sí.

Evankhell561 commented on code/HISPANIA/modules/reagents/reagent_containers/moa.dm:
```diff
@@ -0,0 +1,172 @@
+	else if(istype(W, /obj/item/screwdriver))
+		if(bcell)
+			bcell.loc = get_turf(src.loc)
+			bcell = null
```
ejemplo, de: 
```
		if(bcell)
			bcell.loc = get_turf(src.loc)
			bcell = null
```

a:

```
		if(cell)
			cell.loc = get_turf(src.loc)
			cell = null
```

SagaStars commented:
wHOpSIE DOOPSIEEEE

![](https://cdn.discordapp.com/attachments/387645048386486273/645796715722571805/7130000c6a1bd6d935835712fe68482f.png)

![](https://cdn.discordapp.com/attachments/387645048386486273/645796931477569537/5ca4d22788e5f0b06820f043fbfe98d9.png)
